src/test/java/org/wiremock/integrations/testcontainers/WireMockContainerExtensionsWebhookTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 WireMock Inc, Oleg Nenashev and all project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wiremock.integrations.testcontainers;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.wiremock.integrations.testcontainers.testsupport.http.TestHttpClient;
+import org.wiremock.integrations.testcontainers.testsupport.http.TestHttpServer;
+
+import java.net.http.HttpResponse;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+/**
+ * Tests the WireMock Webhook extension and TestContainers Networking
+ * For this type of tests we should use following steps:
+ * <p>
+ * Use {@link GenericContainer#withAccessToHost(boolean)} to force the host access mechanism
+ * <p>
+ * Use {@link Testcontainers#exposeHostPorts(int...)} to expose host machine ports to containers
+ * <p>
+ * Use {@link GenericContainer#INTERNAL_HOST_HOSTNAME} to calculate hostname for callback
+ *
+ * @see <a href="https://www.testcontainers.org/features/networking/">Testcontainers Networking</a>
+ */
+public class WireMockContainerExtensionsWebhookTest {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(WireMockContainerExtensionsWebhookTest.class);
+    public static final String WIREMOCK_PATH = "/wiremock/callback-trigger";
+    public static final String APPLICATION_PATH = "/application/callback-receiver";
+
+
+    public TestHttpServer applicationServer = TestHttpServer.newInstance();
```

Comment:
Could be another WireMock instance though

Reply:
Let me play with that to see witch one look sbetter